src/web/mod.rs

```diff
@@ -402,25 +398,26 @@ impl Server {
 fn duration_to_str(ts: time::Timespec) -> String {
     let tm = time::at(ts);
     let delta = time::now() - tm;
+    let delta = (
+        delta.num_days(),
+        delta.num_hours(),
+        delta.num_minutes(),
+        delta.num_seconds(),
+    );
 
-    if delta.num_days() > 5 {
-        format!("{}", tm.strftime("%b %d, %Y").unwrap())
-    } else if delta.num_days() > 1 {
-        format!("{} days ago", delta.num_days())
-    } else if delta.num_days() == 1 {
-        "one day ago".to_string()
-    } else if delta.num_hours() > 1 {
-        format!("{} hours ago", delta.num_hours())
-    } else if delta.num_hours() == 1 {
-        "an hour ago".to_string()
-    } else if delta.num_minutes() > 1 {
-        format!("{} minutes ago", delta.num_minutes())
-    } else if delta.num_minutes() == 1 {
-        "one minute ago".to_string()
-    } else if delta.num_seconds() > 0 {
-        format!("{} seconds ago", delta.num_seconds())
-    } else {
-        "just now".to_string()
+    match delta {
+        (days, _, _, _) if days > 5 => format!("{}", tm.strftime("%b %d, %Y").unwrap()),
+        (days, _, _, _) if days > 1 => format!("{} days ago", days),
```

Comment:
```suggestion
        (2..=5, _, _, _) => format!("{} days ago", days),
```

Reply:
Whoops, this again doesn't compile. Try

```suggestion
        (days @ 2..=5, _, _, _) => format!("{} days ago", days),
```